control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -1054,3 +1065,56 @@ impl CreateVolumeExeVal for CreateVolumeSource<'_> {
         }
     }
 }
+
+// spread_label_is_same checks if the node labels is same as that of all existing replicas.
+fn spread_label_is_same(
+    replica: &Replica,
+    replica_candidate: &CreateReplica,
+    context: &Context,
+) -> Result<bool, SvcError> {
+    let Some(ref topology) = context.volume.as_ref().topology else {
+        return Ok(false);
+    };
+
+    let Some(node_topology) = topology.node.as_ref() else {
+        return Ok(false);
+    };
+    let node_exclusion_label_match = match node_topology {
+        NodeTopology::Explicit(_) => true,
+        NodeTopology::Labelled(labelled_topology) => {
+            // If the NodeSpreadTopologyKey is present in the Volume Node Topology, then
+            // get the node details from replica candidate and
+            // check if the node labels is same as that of all existing replicas.
+            // If its same then don't consider the pool for replica creation
+            let mut exclusion_label_match = false;
+            if !labelled_topology.exclusion.is_empty() {
+                let candidate_node_label = context
+                    .registry
+                    .specs()
+                    .node(&replica_candidate.node)?
+                    .labels()
+                    .clone();
```

Comment:
Let's avoid the clone since we only need to compare the labels anyway

Reply:
Marked as resolved but it's not...

Reply:
I tried removing clone().
but there as an errror 
```
error[E0716]: temporary value dropped while borrowed
    --> control-plane/agents/src/bin/core/volume/operations.rs:1091:44
     |
1091 |                   let candidate_node_label = context
     |  ____________________________________________^
1092 | |                     .registry
1093 | |                     .specs()
1094 | |                     .node(&replica_candidate.node)?
     | |___________________________________________________^ creates a temporary value which is freed while still in use
1095 |                       .labels();
     |                                - temporary value is freed at the end of this statement
...
1103 |                           (candidate_node_label.get(key), replica_node_label.get(key))
     |                            -------------------- borrow later used here
     |
     = note: consider using a `let` binding to create a longer lived value
```
since `get` uses this map , i had to use clone.